src/wp-includes/media.php

```diff
@@ -1956,6 +1956,31 @@ function wp_image_use_alternate_mime_types( $image, $context, $attachment_id ) {
 				break;
 			}
 
+			/**
+			 * Filters the image HTML tag.
+			 *
+			 * This filter can be used to replace target mime_type image filename for different sizes
+			 * in the generated <img> HTML tag.
+			 *
+			 * @since 6.1.0
+			 *
+			 * @param array  $image         The HTML `img` tag.
+			 * @param string $attachment_id The attachment ID.
+			 * @param string $metadata      Image size metadata.
+			 * @param string $name          Size identifier - e.g. 'full', 'medium', 'small' etc.
+			 * @param string $target_mime   Image mime type.
+			 * @param string $context       Additional context, like the current filter name or the function name from where this was called.
+			 *
+			 * @return string Updated <img> HTML tag.
+			 */
+			$filtered_image = apply_filters( 'wp_content_pre_replace_additional_image_source', $image, $attachment_id, $metadata, 'full', $target_mime, $context );
+
+			if ( $filtered_image !== $image ) {
+				$image = $filtered_image;
+				unset( $metadata['sources'] );
```

Comment:
why is $metadata['sources'] unset here? should $metadata['sources'] get filled in with the filtered image data?

Reply:
No. `$metadata['sources']` is unset after the image replace below new filter code: https://github.com/adamsilverstein/wordpress-develop-fork/blob/add/plugin-integration-points-55443/src/wp-includes/media.php#L1976-L1988 that's why it unset the `$metadata['sources']`.

Please correct me it am getting anything wrong

Reply:
I see, so this is an optimization, wasn't clear in on the PR, thanks